src/components/FileViewer/FileList.tsx

```diff
@@ -42,11 +56,119 @@ export default function FileList({
+    if (boundingDivRef.current !== null) {
+      setBoundingDivDimensions({ height: boundingDivRef.current.offsetHeight, width: boundingDivRef.current.offsetWidth })
+    }
+  }, [boundingDivRef]);
+
+  useEffect(() => {
+    const handleMouseMove = (e: MouseEvent) => {
+      setMousePos({ x: e.clientX - absCoords.x, y: e.clientY - absCoords.y });
+    };
+
+    window.addEventListener('mousemove', handleMouseMove);
+
+    return () => {
+      window.removeEventListener(
+        'mousemove',
+        handleMouseMove
+      );
+    };
+  }, [absCoords]);
+
+  useEffect(() => {
+    const onResize = () => {
+      if (boundingDivRef.current !== null) {
+        setAbsCoords({
+            x: boundingDivRef.current.getBoundingClientRect().left + window.scrollX, 
+            y: boundingDivRef.current.getBoundingClientRect().top + window.scrollY 
+        });
+        setBoundingDivDimensions({ height: boundingDivRef.current.offsetHeight, width: boundingDivRef.current.offsetWidth })
+      }
+
+      if (contextMenuRef.current !== null) {
+        console.log(contextMenuRef.current.getBoundingClientRect())
+      }
+    }
+
+    if (boundingDivRef.current !== null) {
+      setAbsCoords({
+        x: boundingDivRef.current.getBoundingClientRect().left + window.scrollX, 
+        y: boundingDivRef.current.getBoundingClientRect().top + window.scrollY
+      });
+    }
+
+    window.addEventListener('resize', onResize);
+    return () => {
+        window.removeEventListener('resize', onResize);
+    };
+
+  }, [])
+
+  useEffect(() => {
+    const handleClickOutside = (e: Event) => {
+      if (contextMenuRef.current && !contextMenuRef.current.contains(e.target as Node)) {
+        console.log(contextMenuRef)
+        console.log(contextMenuRef.current.getBoundingClientRect())
+        setShowContextMenu(false)
+      }
+    }
+    window.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      window.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+  
+
+  const calculateContextMenuCoords = () => {
+    let x = null;
+    let y = null;
+    if (mousePos.x + contextMenuDimensions.width > boundingDivDimensions.width) {
+      x = mousePos.x - contextMenuDimensions.width;
+    } else {
+      x = mousePos.x
+    }
+    console.log(boundingDivDimensions.height)
+    if (mousePos.y + contextMenuDimensions.height > boundingDivDimensions.height - 10) {
+      y = boundingDivDimensions.height - contextMenuDimensions.height - 10
+    } else {
+      y = mousePos.y
+    }
+    setContextMenuCoords({ x, y })
+    
+  }
+
   const fileTreeEntryClassName =
     'flex flex-row items-center gap-4 py-2 px-4 text-medium font-medium tracking-medium whitespace-nowrap';
 
   return (
-    <div className="flex h-full w-full grow flex-col @container/file-tree">
+    <div className="flex h-full w-full grow flex-col @container/file-tree" ref={boundingDivRef}>
+      { showContextMenu && 
+        <FileContextMenu 
```

Comment:
changing the z-index didn't seem to work, likely because the `FileList` has a lower z-index than the `FileEditor`, so i instead fixed this by having the context menu fit into the file list box in this case too.
![image](https://user-images.githubusercontent.com/29615067/222634318-6bf57e10-5359-48d8-bcc4-1d02536a9fdc.png)